### PR TITLE
feat(cli): control progress output style with --progress=bar|events

### LINKED ISSUE #8696: [CLI] display issues with new deployment monitoring in the CDK CLI
_Originally posted by @robertd in https://github.com/aws/aws-cdk/pull/8165#issuecomment-647178526_

It's not working really well on iTerm2 with zsh (using robbierussel/oh-my-zsh)?

Is there a way to use more verbose output (~for CI/CD purposes too~)?

```zsh
➜  sipp-api-cdk git:(master) ✗ cdk deploy
SippApiStack: deploying...
SippApiStack: creating CloudFormation changeset...








 ✅  SippApiStack

Outputs:
SippApiStack.ServiceServiceURL250C0FB6 = https://sipp-api-alb.acme.com
SippApiStack.ServiceLoadBalancerDNSEC5B149E = internal-SippA-Servi-1X2ZSXSR3W328-1234567890.us-west-2.elb.amazonaws.com

Stack ARN:
arn:aws:cloudformation:us-west-2:1234567890:stack/SippApiStack/3841b3e0-b271-11ea-aae2-065639e88e5c
```

.
.
.

**edit**: Luckily everything looks fine in Gitlab's CI/CD pipeline.

image

### PR DESCRIPTION
**Pull request by shivlaks**
Since the new deployment monitoring has been problematic on environments such as CI,
adding a switch called `progress` to allow users to set their deploy output display preference.

Supported values are `bar` and `events`. The default deployment display mode includes the
progress bar.

The workaround of adding `-v` to display the stack event history includes a lot of
undesired output and users have expressed that it includes a lot of noise.

If the key `progress` exists in user settings (`~/.cdk.json`) or project settings (`cdk.json`),
users will not need to specify it on every deploy and can set their desired default once.

closes #8696 

----

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

### CODE REVIEW

**Comment by eladb** [comment]
To improve discoverability of the various options and allow future modes let's use an enumeration:

```
--progress=bar (current behavior)
--progress=events (old behavior)
```

File: packages/aws-cdk/lib/api/util/cloudformation/stack-activity-monitor.ts [hunk 1]
```diff
@@ -17,6 +17,21 @@ interface ResourceMetadata {
...
 }
 
+/**
+ * Supported display modes for stack deployment activity
+ */
+export enum StackActivityProgress {
+  /**
+   * Displays a progress bar with only the events for the resource currently being deployed
+   */
+  Bar = 'bar',
```
Comment (eladb) [inline packages/aws-cdk/lib/api/util/cloudformation/stack-activity-monitor.ts:27]:
```suggestion
  BAR = 'bar',
```

**Review by eladb — APPROVED** [review]
Update PR description to reflect updated switch name

**Comment by eladb** [comment]
Added do-not-merge

**Comment by farzad-xgen** [comment]
```$ cdk --help``` doesn't show this flag.

**Comment by robertd** [comment]
@farzad-xgen this flag is under `cdk deploy`.

**Comment by farzad-xgen** [comment]
Got it, thanks!

**Comment by robertd** [comment]
@shivlaks  Can you please provide an example on how to set this in `~/.cdk.json` globally? I can't seem to get it right.

Thanks!

**Comment by robertd** [comment]
@shivlaks poke

**Pull request merged**